nixos/tests/web-servers/static-web-server.nix

```diff
@@ -0,0 +1,27 @@
+import ../make-test-python.nix ({ pkgs, lib, ...} : {
+  name = "static-web-server";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ mac-chaffee ];
+  };
+
+  nodes.machine = { pkgs, ... }: {
+    services.static-web-server = {
+      enable = true;
+      listen = "[::]:8080";
+      root = "/tmp/";
+      configuration = {
+        general = { directory-listing = true; };
+      };
+    };
+  };
+
+  testScript = ''
```

Comment:
would be nice if this also tests the files served, perhaps if we change `root` to a store path